multiapps-controller-persistence/src/main/java/org/cloudfoundry/multiapps/controller/persistence/services/ProcessLogger.java

```diff
@@ -1,84 +1,77 @@
 package org.cloudfoundry.multiapps.controller.persistence.services;
 
-import java.io.File;
-
 import org.apache.commons.io.FileUtils;
-import org.apache.log4j.Logger;
+import org.apache.logging.log4j.core.Logger;
+import org.apache.logging.log4j.core.LoggerContext;
 
-public class ProcessLogger extends Logger {
+import java.io.File;
 
-    private static final String NULL_LOGGER_NAME = "Null logger";
+public class ProcessLogger {
 
     private Logger logger;
     private File log;
     private String logName;
     protected final String spaceId;
     protected final String processId;
     protected final String activityId;
+    private LoggerContext loggerContext;
 
-    public ProcessLogger(Logger logger, File log, String logName, String spaceId, String processId, String activityId) {
-        super(logger.getName());
+    public ProcessLogger(LoggerContext loggerContext, Logger logger, File log, String logName, String spaceId,
+                         String processId, String activityId) {
         this.logger = logger;
         this.log = log;
         this.logName = logName;
         this.spaceId = spaceId;
         this.processId = processId;
         this.activityId = activityId;
+        this.loggerContext = loggerContext;
     }
 
-    public ProcessLogger(String spaceId, String processId, String activityId) {
-        super(NULL_LOGGER_NAME);
+    public ProcessLogger(LoggerContext loggerContext, String spaceId, String processId, String activityId) {
+        this.loggerContext = loggerContext;
         this.spaceId = spaceId;
         this.processId = processId;
         this.activityId = activityId;
     }
 
-    @Override
     public void info(Object message) {
         logger.info(message);
     }
 
-    @Override
     public void debug(Object message) {
         logger.debug(message);
     }
 
-    @Override
+    public void debug(Object message, Throwable throwable) {
+        logger.debug(message, throwable);
+    }
+
     public void error(Object message) {
         logger.error(message);
     }
 
-    @Override
     public void error(Object message, Throwable t) {
         logger.error(message, t);
     }
 
-    @Override
     public void trace(Object message) {
         logger.trace(message);
     }
 
-    @Override
     public void warn(Object message) {
         logger.warn(message);
     }
 
-    @Override
     public void warn(Object message, Throwable t) {
         logger.warn(message, t);
     }
 
-    @Override
-    public synchronized void removeAllAppenders() {
-        logger.removeAllAppenders();
-    }
-
     public String getProcessId() {
-        return processId;
+        return this.processId;
     }
 
     public String getActivityId() {
-        return activityId;
+        return this.activityId;
```

Comment:
What is difference between activity and process id?

Reply:
The process id also refers to correlation id and it shows unique identifier of the particular process. On the other hand activity id refers to the task id and identifies the unique identifier of the particular task that has been executed.